share/completions/s3cmd.fish

```diff
@@ -0,0 +1,178 @@
+# s3cmd
+
+function __s3cmd_is_remote_path
+	commandline -pct | string match -r -- "^s3://.*"
+end
+
+function __s3cmd_is_valid_remote_path
+	commandline -pct | string match -r -- "^s3://.*?/.*"
+end
+
+function __prev_tokens
+	set __tokens (commandline -pco)
+	for arg in $argv
+		if echo $__tokens[-1] | string match $arg
+			return 0
+		end
+	end
+	return 1
+end
+
+function __initial_command
+	set __tokens (commandline -pco)
+	string match (count $__tokens) 1
```

Comment:
There is also `__fish_is_first_token` already. That one skips options.

Personally, for subcommands, I prefer `not __fish_seen_subcommand_from $cmds` (see e.g. the systemctl completions). That allows for options and option-arguments, as long as they don't look like a command.

For a completely correct solution, you're going to need to do something like the git completions, or use argparse (which I'm converting both the git and systemctl completions to).